internal/provider/data_source_dirhash_sha256.go

```diff
@@ -35,26 +35,24 @@ func dataSourceDirhashSha256() *schema.Resource {
 	}
 }
 
-func dataSourceDirhashSha256Read(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	var dir = d.Get("directory").(string)
-	d.SetId(dir)
-	var ignore = []string{}
-	if v, ok := d.GetOk("ignore"); ok {
-		ignore = stringListToStringSlice(v.([]interface{}))
+func dataSourceDirhashSha256Read(ctx context.Context, dr *schema.ResourceData, meta any) diag.Diagnostics {
+	dir := dr.Get("directory").(string)
+	dr.SetId(dir)
+	var ignore []string
+	if v, ok := dr.GetOk("ignore"); ok {
+		ignore = stringInterfaceListToStringSlice(v.([]interface{}))
 	}
-	var checksum = lib.DirHash(dir, ignore)
-	d.Set("checksum", checksum)
+	checksum := lib.DirHash(dir, ignore)
+	dr.Set("checksum", checksum)
 	return nil
 }
 
-func stringListToStringSlice(stringList []interface{}) []string {
-	ret := []string{}
+func stringInterfaceListToStringSlice(stringList []interface{}) []string {
+	var ret []string
 	for _, v := range stringList {
-		if v == nil {
-			ret = append(ret, "")
-			continue
+		if v != nil {
+			ret = append(ret, v.(string))
```

Comment:
are we really sure that only interface of type string can reach here? failing that the code will panic.
example from go playground:
```go
package main

import "fmt"

func main() {
	var d interface{}
	d = 45
	fmt.Println(d.(string))
}
```

when running the code above it panics:
```
panic: interface conversion: interface {} is int, not string

goroutine 1 [running]:
main.main()
	/tmp/sandbox615016875/prog.go:10 +0x2e

Program exited.
```

Reply:
Yes, we are @NabilHouidi Terraform generates it that way based on the schema